autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -366,10 +374,10 @@ def _check_data(
                 self.column_order = column_order
 
             dtypes = [dtype.name for dtype in X.dtypes]
-
+            diff_cols = X.columns[[s_dtype != dtype for s_dtype, dtype in zip(self.dtypes, dtypes)]]
             if len(self.dtypes) == 0:
                 self.dtypes = dtypes
-            elif self.dtypes != dtypes:
+            elif not self._is_datasets_consistent(diff_cols, X):
```

Comment:
I have not understood how you are working on this PR, but do we need `_is_datasets_consistent` after all?
I thought that you changed files in a way that we need to check only dtype differences.

Reply:
hey, sorry for this. Actually, the issue I'm facing is that this PR is not running any checks. So the changes I made were causing some tests to fail. I ran the tests locally and found that we still need that check.